metadata-ingestion/src/datahub/sdk/mlmodel.py

```diff
@@ -0,0 +1,328 @@
+
+    @property
+    def description(self) -> Optional[str]:
+        return self._ensure_model_props().description
+
+    def set_description(self, description: str) -> None:
+        self._ensure_model_props().description = description
+
+    @property
+    def external_url(self) -> Optional[str]:
+        return self._ensure_model_props().externalUrl
+
+    def set_external_url(self, external_url: str) -> None:
+        self._ensure_model_props().externalUrl = external_url
+
+    @property
+    def custom_properties(self) -> Optional[Dict[str, str]]:
+        return self._ensure_model_props().customProperties
+
+    def set_custom_properties(self, custom_properties: Dict[str, str]) -> None:
+        self._ensure_model_props().customProperties = custom_properties
+
+    @property
+    def created(self) -> Optional[datetime]:
+        return parse_time_stamp(self._ensure_model_props().created)
+
+    def set_created(self, created: datetime) -> None:
+        self._ensure_model_props().created = make_time_stamp(created)
+
+    @property
+    def last_modified(self) -> Optional[datetime]:
+        return parse_time_stamp(self._ensure_model_props().lastModified)
+
+    def set_last_modified(self, last_modified: datetime) -> None:
+        self._ensure_model_props().lastModified = make_time_stamp(last_modified)
+
+    @property
+    def training_metrics(self) -> Optional[List[MLMetricClass]]:
+        return self._ensure_model_props().trainingMetrics
+
+    def set_training_metrics(
+        self, metrics: Union[List[MLMetricClass], Dict[str, Any]]
+    ) -> None:
+        self._ensure_model_props().trainingMetrics = self._convert_training_metrics(
+            metrics
+        )
+
+    def add_training_metric(self, name: str, value: Any) -> None:
+        props = self._ensure_model_props()
+        if props.trainingMetrics is None:
+            props.trainingMetrics = []
+        props.trainingMetrics.append(MLMetricClass(name=name, value=str(value)))
+
+    @property
+    def hyper_params(self) -> Optional[List[MLHyperParamClass]]:
+        return self._ensure_model_props().hyperParams
+
+    def set_hyper_params(
+        self, params: Union[List[MLHyperParamClass], Dict[str, Any]]
+    ) -> None:
+        self._ensure_model_props().hyperParams = self._convert_hyper_params(params)
+
+    def add_hyper_param(self, name: str, value: Any) -> None:
+        props = self._ensure_model_props()
+        if props.hyperParams is None:
+            props.hyperParams = []
+        props.hyperParams.append(MLHyperParamClass(name=name, value=str(value)))
+
+    @property
+    def version(self) -> Optional[str]:
+        version_props = self._get_aspect(VersionPropertiesClass)
+        if version_props and version_props.version:
+            return version_props.version.versionTag
+        return None
+
+    @property
+    def aliases(self) -> Optional[List[str]]:
+        version_props = self._get_aspect(VersionPropertiesClass)
```

Comment:
is VersionPropertiesClass a standard aspect that applies across entity types? if so, should it be set up as a mixin type similar to how the rest of the tags/owners/terms/etc are done?

Reply:
done